macros/src/main/scala/com/avsystem/commons/macros/serialization/GenCodecUtilMacros.scala

```diff
@@ -0,0 +1,47 @@
+package com.avsystem.commons
+package macros.serialization
+
+import scala.annotation.tailrec
+import scala.reflect.macros.blackbox
+
+class GenCodecUtilMacros(ctx: blackbox.Context) extends CodecMacroCommons(ctx) {
```

Comment:
Do you think this macro could share some code with GenCodec macros for more intended coupling now that they are in the same module?